src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -488,20 +492,36 @@ void ProtocolHandlerImpl::SendStartSessionNAck(
 
   uint8_t maxProtocolVersion = SupportedSDLProtocolVersion();
 
+  utils::SemanticVersion full_version;
+  session_observer_.ProtocolVersionUsed(
```

Comment:
are we sure the connection is valid when this code is run? we don't need to check the return value of ProtocolVersionUsed? If an app sends EndSession and stops running, maybe it's possible the connection could be removed by now.

Reply:
Good call. Checks added in https://github.com/smartdevicelink/sdl_core/pull/3462/commits/c785dfa7661aba9511513c072b9ce8ef26ef1cbc